lib/java/com/google/android/material/divider/MaterialDivider.java

```diff
@@ -0,0 +1,193 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.material.divider;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.util.AttributeSet;
+import android.util.TypedValue;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewGroup.LayoutParams;
+
+import androidx.annotation.ColorInt;
+import androidx.annotation.Dimension;
+import androidx.annotation.Px;
+
+import com.google.android.material.R;
+import com.google.android.material.internal.ThemeEnforcement;
+import com.google.android.material.internal.ViewUtils;
+
+import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;
+import static android.view.ViewGroup.LayoutParams.WRAP_CONTENT;
+import static com.google.android.material.theme.overlay.MaterialThemeOverlay.wrap;
+
+public class MaterialDivider extends View {
+
+
+  public MaterialDivider(Context context) {
+    this(context, null);
+  }
+
+  public MaterialDivider(Context context, AttributeSet attrs) {
+    this(context, attrs, R.attr.dividerStyle);
+  }
+
+  boolean isRtlEnabled;
+
+  AttributeSet initialAttrs;
+  int initialDefStyleAttr;
+
+  private static final int DEF_STYLE_RES = R.style.Widget_MaterialComponents_Divider;
+
+  public MaterialDivider(Context context, AttributeSet attrs, int defStyleAttr) {
+
+    super(wrap(context, attrs, defStyleAttr, DEF_STYLE_RES), attrs, defStyleAttr);
+    initialAttrs = attrs;
+    initialDefStyleAttr = defStyleAttr;
+    isRtlEnabled = ViewUtils.isLayoutRtl(this);
+
+    float alpha = attrs.getAttributeFloatValue("http://schemas.android.com/apk/res/android", "alpha", -1);
+    int color = attrs.getAttributeIntValue(android.R.attr.background, -1);
+
+    final TypedArray dividerStyleAttrs = getDividerStyleAttrs();
+    if (alpha == -1) {
+      alpha = dividerStyleAttrs.getFloat(ATTR_ALPHA, -1);
+    }
+    if (color == -1) {
+      color = dividerStyleAttrs.getColor(ATTR_COLOR, -1);
+    }
+
+    setAlpha(alpha);
+    setColor(color);
+  }
+
+  int[] ATTRS = {R.attr.dividerAlpha, R.attr.dividerColor, R.attr.dividerInset, R.attr.dividerSize};
+  private static final int ATTR_ALPHA = 0;
+  private static final int ATTR_COLOR = 1;
+  private static final int ATTR_INSET = 2;
+  private static final int ATTR_SIZE = 3;
+
+  private TypedArray getDividerStyleAttrs() {
+    TypedValue dividerStyle = new TypedValue();
+    getContext().getTheme().resolveAttribute(R.attr.dividerStyle, dividerStyle, true);
+    return getContext().obtainStyledAttributes(dividerStyle.resourceId, ATTRS);
+  }
+
+  //Runs when this view's layoutParams are available
+  @Override
+  protected void onAttachedToWindow() {
+
+    super.onAttachedToWindow();
+    initializeOrientation();
+    TypedArray dividerStyleAttrs = getDividerStyleAttrs();
```

Comment:
Getting styles attributes should be done exclusively in the constructor (as I've done above), not here